typings/solid-namespace/index.d.ts

```diff
@@ -1,8 +1,9 @@
 declare module 'solid-namespace' {
   import { NamedNode } from 'rdflib'
 
+  type RDFLibSubset = { namedNode: (value: string) => NamedNode }
   type toNamedNode = (label: string) => NamedNode
   export type Namespaces = {[alias: string]: toNamedNode}
 
-  export default function vocab(rdflib: typeof import('rdflib')): Namespaces
+  export default function vocab(rdf: RDFLibSubset): Namespaces
```

Comment:
You likely want to rely on the `DataFactory` export of `rdflib`, which has standard RDF/JS typings.

Reply:
I will do this after fixing the typings for rdflib, which exposes DataFactory as a namespace (which cannot be used as specific type in TypeScript).

Reply:
Actually, based on this, maybe we should try to expose typings file in rdflib? So we don't have to go through https://github.com/DefinitelyTyped/DefinitelyTyped to make these changes. [I added an issue on rdflib for this](https://github.com/linkeddata/rdflib.js/issues/341).